src/tools/perf/lib/uct_tests.cc

```diff
@@ -783,21 +783,50 @@ class uct_perf_test_runner {
    TEST_CASE_ALL_OSD(_perf, _case, UCT_PERF_DATA_LAYOUT_BCOPY) \
    TEST_CASE_ALL_OSD(_perf, _case, UCT_PERF_DATA_LAYOUT_ZCOPY)
 
+#define DISPATCH(_name, _cmd, _type) \
+static ucs_status_t uct_perf_dispatch_##_name(ucx_perf_context_t *perf) { \
+    UCS_PP_FOREACH(TEST_CASE_ALL_DATA, perf, \
+        (_cmd, _type) \
+    ); \
+    return UCS_ERR_INVALID_PARAM; \
+}
+
+#define DISPATCHER_ENTRY(_name, _cmd, _type) uct_perf_dispatch_##_name,
+
+#define TEST_CASES(_macro) \
+    _macro(pingpong_am,     UCX_PERF_CMD_AM,    UCX_PERF_TEST_TYPE_PINGPONG)   \
+    _macro(pingpong_put,    UCX_PERF_CMD_PUT,   UCX_PERF_TEST_TYPE_PINGPONG)   \
+    _macro(pingpong_add,    UCX_PERF_CMD_ADD,   UCX_PERF_TEST_TYPE_PINGPONG)   \
+    _macro(stream_uni_am,   UCX_PERF_CMD_AM,    UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_put,  UCX_PERF_CMD_PUT,   UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_get,  UCX_PERF_CMD_GET,   UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_add,  UCX_PERF_CMD_ADD,   UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_fadd, UCX_PERF_CMD_FADD,  UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_swap, UCX_PERF_CMD_SWAP,  UCX_PERF_TEST_TYPE_STREAM_UNI) \
+    _macro(stream_uni_cswap,UCX_PERF_CMD_CSWAP, UCX_PERF_TEST_TYPE_STREAM_UNI)
+
+TEST_CASES(DISPATCH)
+
+typedef ucs_status_t (*uct_dispatch_func_t)(ucx_perf_context_t *perf);
+
+static uct_dispatch_func_t dispatchers[] = {
+    TEST_CASES(DISPATCHER_ENTRY)
+};
```

Comment:
maybe define this variable inside uct_perf_test_dispatch (still as static)?

Reply:
done